lib/ecto/adapters/sql.ex

```diff
@@ -1377,21 +1425,29 @@ defmodule Ecto.Adapters.SQL do
 
   @repo_modules [Ecto.Repo.Queryable, Ecto.Repo.Schema, Ecto.Repo.Transaction]
 
-  defp last_non_ecto([{mod, _, _, _} | _stacktrace], repo, last)
+  def first_non_ecto_stacktrace(stacktrace, repo, size) do
```

Comment:
We need to document this function. But also document the new option in `Ecto`. But this function could have a full example of how to configure it and use it :)